src/TestStack.White/InputDevices/Keyboard.cs

```diff
@@ -1,193 +1,230 @@
 using TestStack.White.WindowsAPI;
 using Action = TestStack.White.UIItems.Actions.Action;
 
 namespace TestStack.White.InputDevices
 {
-    //BUG: KeysConverter
+    // BUG: KeysConverter
     /// <summary>
     /// Represents Keyboard attachment to the machine.
     /// </summary>
     public class Keyboard : IKeyboard
     {
-        private static readonly List<KeyboardInput.SpecialKeys> scanCodeDependent = new List<KeyboardInput.SpecialKeys>
-                                                                               {
-                                                                                   KeyboardInput.SpecialKeys.RIGHT_ALT,
-                                                                                   KeyboardInput.SpecialKeys.INSERT,
-                                                                                   KeyboardInput.SpecialKeys.DELETE,
-                                                                                   KeyboardInput.SpecialKeys.LEFT,
-                                                                                   KeyboardInput.SpecialKeys.HOME,
-                                                                                   KeyboardInput.SpecialKeys.END,
-                                                                                   KeyboardInput.SpecialKeys.UP,
-                                                                                   KeyboardInput.SpecialKeys.DOWN,
-                                                                                   KeyboardInput.SpecialKeys.PAGEUP,
-                                                                                   KeyboardInput.SpecialKeys.PAGEDOWN,
-                                                                                   KeyboardInput.SpecialKeys.RIGHT,
-                                                                                   KeyboardInput.SpecialKeys.LWIN,
-                                                                                   KeyboardInput.SpecialKeys.RWIN
-                                                                               };
-
         private readonly List<KeyboardInput.SpecialKeys> heldKeys = new List<KeyboardInput.SpecialKeys>();
+        protected readonly List<int> KeysHeld = new List<int>();
 
         /// <summary>
-        /// Use Window.Keyboard method to get handle to the Keyboard. Keyboard instance got using this method would not wait while the application
-        /// is busy.
+        /// Use Window.Keyboard method to get handle to the Keyboard. 
+        /// Keyboard instance got using this method would not wait while the application is busy.
         /// </summary>
-        public static readonly Keyboard Instance = new Keyboard();
-
-        private readonly List<int> keysHeld = new List<int>();
+        [Obsolete("Obsolete. Create an instance where it is needed.")]
+        public static readonly IKeyboard Instance = new Keyboard();
```

Comment:
I am very torn on this.
What I do not want is a mixture of the two possibilities.

On one side, Mouse and Keyboard are things that can be accessed via a static instance.
Since Mouse and Keyboard are the same system wide.

On the other hand, it could make sense to allow Users to create their own Mouse and Keyboard implementations that adhere to the interface but with a custom workflow inside.

If you look at unit testing, persistent states are really bad when it comes to testing since tests should IDEALLY be independent of each other.

I would like to postpone this to a later point to let some more thoughts go into it.